src/operator/leaky_relu-inl.h

```diff
@@ -332,166 +332,51 @@ class LeakyReLUOp : public Operator {
+    LeakyReLUOp<xpu, DType> op(param);
+    op.Forward(ctx, inputs, req, outputs, no_use_but_adapt_origin_api);
+  });
+}
 
-  std::vector<std::pair<int, void*> > BackwardInplaceOption(
-    const std::vector<int> &out_grad,
-    const std::vector<int> &in_data,
-    const std::vector<int> &out_data,
-    const std::vector<void*> &in_grad) const override {
-    return {{out_grad[leakyrelu::kOut], in_grad[leakyrelu::kData]}};
-  }
-
-  std::vector<std::pair<int, void*> > ForwardInplaceOption(
-    const std::vector<int> &in_data,
-    const std::vector<void*> &out_data) const override {
-    if (param_.act_type == leakyrelu::kPReLU) {
-      return {};
-    } else {
-      return {{in_data[leakyrelu::kData], out_data[leakyrelu::kOut]}};
-    }
-  }
-
-  std::vector<std::string> ListArguments() const override {
-    if (param_.act_type == leakyrelu::kPReLU) {
-      return {"data", "gamma"};
-    } else {
-      return {"data"};
-    }
-  }
-
-  std::vector<std::string> ListOutputs() const override {
-    if (param_.act_type == leakyrelu::kRReLU) {
-      return {"output", "mask"};
-    } else {
-      return {"output"};
-    }
-  }
-
-  int NumOutputs() const override {
-    if (param_.act_type == leakyrelu::kRReLU) {
-      return 2;
-    } else {
-      return 1;
-    }
-  }
-
-  int NumVisibleOutputs() const override {
-    return 1;
-  }
-
-  std::vector<ResourceRequest> ForwardResource(
-      const mxnet::ShapeVector &in_shape) const override {
-    if (param_.act_type == leakyrelu::kRReLU) {
-      return {ResourceRequest::kRandom};
-    } else {
-      return std::vector<ResourceRequest>();
-    }
-  }
+template<typename xpu>
+void LeakyReLUGradCompute(const nnvm::NodeAttrs& attrs,
+                          const OpContext& ctx,
+                          const std::vector<TBlob>& inputs,
+                          const std::vector<OpReqType>& req,
+                          const std::vector<TBlob>& outputs) {
+  const LeakyReLUParam& param = nnvm::get<LeakyReLUParam>(attrs.parsed);
+  const std::vector<TBlob> no_use_but_adapt_origin_api;
+  // inputs: out_grad, input_data, input_gamma, output, output_mask
+  size_t expected_in = param.act_type == leakyrelu::kPReLU ? 2 : 1;
+  size_t expected_out = param.act_type == leakyrelu::kRReLU ? 2 : 1;
 
-  std::vector<ResourceRequest> BackwardResource(
-      const mxnet::ShapeVector &in_shape) const override {
-    return {ResourceRequest::kTempSpace};
-  }
+  std::vector<TBlob> out_grad{inputs[0]};
+  std::vector<TBlob> in_data(inputs.begin() + 1,
+                             inputs.begin() + 1 + expected_in);
+  std::vector<TBlob> out_data(inputs.begin() + 1 + expected_in,
+                              inputs.begin() + 1 + expected_in + expected_out);
 
-  Operator* CreateOperator(Context ctx) const override {
-    LOG(FATAL) << "Not Implemented.";
-    return NULL;
-  }
+  CHECK_EQ(out_grad.size(), 1U);
+  CHECK_EQ(req.size(), expected_in);
+  CHECK_EQ(in_data.size(), expected_in);
```

Comment:
No need to check the size of out_grad and in_data. They're created in this function.
`CHECK_EQ(req.size(), outputs.size());`